1020-progress-bar-design-guidance.md

```diff
@@ -0,0 +1,31 @@
+---
+category: component design guidance
+state: draft
+created: 2024-10-16
+updated: 2024-10-22
+---
+
+# Progress Bar Design Guidance
+
+A [progress bar](https://clarity.design/documentation/progress) is a linear indicator that shows the progress of an ongoing, user-initiated process. Progress bars are designed for use in the [main content area](https://clarity.design/documentation/app-layout), [header](https://clarity.design/documentation/header), [cards](https://clarity.design/documentation/card), and [modals](https://clarity.design/documentation/modal).
+
+## Guidance
+
+### Determinate Progress Bar
+
+- **Must** use "determinate progress bar" when the process has a known duration. Examples:
+    - file downloads or uploads
+    - a series of configuration steps that happen during an initialization process
+- **Must** explain what the progress bar is showing and why it might take time. For example, if you're uploading a file, provide the file name and size.
```

Comment:
```suggestion
- Design teams **must** explain what the progress bar is showing and why it might take time. For example, if uploading a file, include details such as the file name and size. This transparency helps users understand the ongoing process and sets appropriate expectations for completion time.
```